packages/react-server/src/ReactFlightServer.js

```diff
@@ -452,10 +458,6 @@ export function resolveModelToJSON(
   switch (value) {
     case REACT_ELEMENT_TYPE:
       return '$';
-    case REACT_LAZY_TYPE:
-      throw new Error(
-        'React Lazy Components are not yet supported on the server.',
-      );
```

Comment:
@sebmarkbage 

In testing I could not ever get this case to throw. the value at this point is a React.Element and not a symbol so it ends up throwing in `attemptResolveElement` instead. If I am missing a use case that did hit this path that needs to be covered by tests let me know

Reply:
You can now use lazy in the position of React "nodes" (i.e. children). That's actually what Flight itself does on the client.

```js
const element = React.lazy(async () => {
  return <Component prop="prop" />;
});
...
<div>{element}</div>
```

Reply:
Also technically, you could pass just the symbol. `<Component foo={lazyComponent.$$typeof} />`

Regardless, if we see it, it's probably a bug.

Reply:
I'll add support for it, thanks

Reply:
What should we do if someone does this
```
const ElementDisguisedAsAComponent = React.lazy(async () => {
  return <Component prop={"original"} />;
});
...
<div><ElementDisguisedAsAComponent prop={"new"}</div>
```

feels like that should error in Dev and in prod we should render the actual element throwing away the "new" prop?

Reply:
What happens in the client when you do this?

Reply:
The componentDisguisedAsElement case seems like it should hit this case?

https://github.com/facebook/react/blob/main/packages/react-server/src/ReactFlightServer.js#L635

Regardless, it should behave as if there was no lazy around it.